tests/output/test_pvd_output.py

```diff
@@ -121,7 +121,9 @@ def test_append(mesh, tmpdir):
     restarted_outfile = File(str(tmpdir.join("restart_test.pvd")), mode="a")
     restarted_outfile.write(g)
 
-    files_in_tmp = [f for f in listdir(str(tmpdir)) if isfile(join(str(tmpdir), f))]
+    vtufile_dir = tmpdir.join("restart_test")
+    files_in_tmp = [f for f in listdir(str(tmpdir)) if isfile(join(str(tmpdir), f))] \
+        + [f for f in listdir(str(vtufile_dir)) if isfile(join(str(vtufile_dir), f))]
```

Comment:
I think it might make sense to split this into two checks. You are looking for files in different directories and this isn't at all obvious using this code.